MINOR: [Website] Reword ADBC announcement (#299)
apache/arrow-site: _posts/2023-01-05-introducing-arrow-adbc.md

  Libraries like [Turbodbc][turbodbc] and [arrow-jdbc][arrow-jdbc] handle row-to-columnar conversions for clients.
  But this doesn't fundamentally solve the problem.
  Unnecessary data conversions are still required.
- *Directly use database protocols*.
  For some databases, applications can use a database protocol or SDK to directly get Arrow data.

>   For example, applications could use be written with [Arrow Flight SQL][flight-sql] to connect to Dremio and other databases that support the Flight SQL protocol.

Reviewer: ```suggestion
  For example, applications could use [Arrow Flight SQL][flight-sql] to connect to Dremio and other databases that support the Flight SQL protocol.
```

(If you want, I think it's fair to link "Dremio" to the website as well.)

>   But not all databases support the Flight SQL protocol. An example is Google BigQuery, which has a separate SDK that returns Arrow data. In this case, client applications that want to support additional protocols would need to integrate with each of them.

Reviewer: ```suggestion
  But not all databases support Flight SQL, even if they support Arrow data. An example is Google BigQuery, which has a separate SDK that returns Arrow data. In this case, client applications that want to support additional databases would need to integrate with each of their protocols.
```

    <tr>
      <th></th>
      <th class="align-top" style="width: 40%" scope="col">Vendor-neutral (database APIs)</th>

>       <th class="align-top" style="width: 40%" scope="col">Database protocols</th>

Reviewer: I think it's still fair to call them vendor-specific; after all, multiple databases also use the PostgreSQL protocol (it just doesn't have a generic name). Maybe "varies by vendor (database protocols)"?

Reviewer 2: Maybe a better phrase would be "Database specific protocols"
